File: javascript/UsernameValidator.js
class UsernameValidator {
  isValid(username) {
    if (username === null || username === undefined || username === '') {
      return false;
    }
    if (username.length < 3 || username.length > 20) {
      return false;
    }
    if (!/^[a-zA-Z]/.test(username)) {
      return false;
    }
    if (!/^[a-zA-Z][a-zA-Z0-9_]*$/.test(username)) {
      return false;
    }
    return true;
  }
}

module.exports = UsernameValidator;

File: java/src/main/java/UsernameValidator.java
public class UsernameValidator {
    public boolean isValid(String username) {
        if (username == null || username.isEmpty()) {
            return false;
        }
        if (username.length() < 3 || username.length() > 20) {
            return false;
        }
        char first = username.charAt(0);
        if (!Character.isLetter(first)) {
            return false;
        }
        for (int i = 1; i < username.length(); i++) {
            char ch = username.charAt(i);
            if (!Character.isLetterOrDigit(ch) && ch != '_') {
                return false;
            }
        }
        return true;
    }
}
